File: baseSrc/h5/pan3d/engine/skill/key/SkillEffect.ts
module Pan3d {
    export class SkillEffect extends SkillKey {
        private _active: Object3D;
        public set active(value: Object3D) {
            this._active = value;
        }
        public get active(): Object3D {
            return this._active;
        }

        public addToRender(): void {
            super.addToRender();
            this.particle.addEventListener(BaseEvent.COMPLETE, this.onPlayCom, this);
        }

        protected onPlayCom(event: Event = null): void {
            this.particle.removeEventListener(BaseEvent.COMPLETE, this.onPlayCom, this);
            ParticleManager.getInstance().removeParticle(this.particle);
            this.removeCallFun(this);
        }


    }
}

File: test/four/app/src/main/java/com/z3d/skill/key/SkillEffect.java
package com.z3d.skill.key;

import com.z3d.base.Object3D;
import com.z3d.event.BaseEvent;
import com.z3d.event.EventCallBack;
import com.z3d.filemodel.ParticleManager;
import com.z3d.scene.Scene3D;

public class SkillEffect extends SkillKey{

    private static final String TAG = "SkillEffect";
    public Object3D active;

    public SkillEffect(Scene3D val) {
        super(val);
    }


    public void addToRender(ParticleManager val) {
        super.addToRender(val);
         this.particle.addEventListener(BaseEvent.COMPLETE,this.getOnPlayCom() , this);
    }
    EventCallBack onPlayCom;
    private EventCallBack getOnPlayCom(){
        if(onPlayCom!=null){
            return onPlayCom;
        }
        onPlayCom=new EventCallBack() {
            @Override
            public void call(Object val, BaseEvent event) {
               particle.removeEventListener(BaseEvent.COMPLETE,getOnPlayCom(),this);
                scene3D.particleManager.removeParticle(particle);
                removeCallFun.StateChange(this);
            }
        };
        return onPlayCom;
    }


}
